### ISSUE TITLE
Improve new schemes color readability

### ISSUE THREAD

**Issue by maclockard** [material-theme/vsc-material-theme#412]
The new style of syntax highlighting looks drastically different from what it used to look like.  

Before:
<img width="583" alt="Screen Shot 2020-01-06 at 2 03 17 PM" src="https://user-images.githubusercontent.com/7308388/71852359-61f21b00-308d-11ea-9801-4fa7d8dc8852.png">

Now:
<img width="572" alt="Screen Shot 2020-01-06 at 2 03 37 PM" src="https://user-images.githubusercontent.com/7308388/71852370-67e7fc00-308d-11ea-9f90-f4c11486e9c9.png">

I'm fine with some aspects of syntax highlighting changing to help increase maintainability, but this is a bit much for me. The older one was more readable and had fewer extreme color changes. For example red/green are complementary colors and have them right next to each other in the import statements is a bit jarring. Compare this with the older highlighting which used blue/green which is a much smoother transition.

Would it be possible to have the new highlighting look more like the legacy highlighting?

I would just use the legacy highlighting, but I'm concerned since in the release notes it was explicitly called out that the legacy highlighting is deprecated and will be deleted at some point.

<img width="892" alt="Screen Shot 2020-01-06 at 2 15 27 PM" src="https://user-images.githubusercontent.com/7308388/71853028-16407100-308f-11ea-8dbb-24fc9abcbb22.png">

I really liked this extension and how the syntax highlighting paired so well with the main theme, it makes me sad to see such a drastic change.

**Additional context**

#411 #410 are some old issues on this same topic and bring up some other concerns

**Comment by maclockard** [comment]
sorry for creating a new issue to add these comments, its just that they old issues got locked down pretty quickly so I couldn't post a response. I really appreciate the time y'all put into maintaining this theme!

**Comment by equinusocio** [comment]
The red/green pairing is something we can work on, but new themes are meant to be **new** and different from older schemes, they will never look the same but we can fix/improve color pairing with new schemes.

**Comment by maclockard** [comment]
@equinusocio If you are not planning on making the new highlighting scheme look similar to the legacy scheme would you be willing to change your stance on deleting the legacy schemes? I understand not maintaining them, but it would make me sad if they were deleted.

**Comment by jhpratt** [comment]
@equinusocio If it's meant to be a new theme, wouldn't it be better to place it under a new name?

**Comment by SharpMobileCode** [comment]
+1 for not deleting the Legacy themes.  I say keep them indefinitely.  If you choose to not maintain them, I understand as we are all software devs here.  I'm not digging the new themes, but am open to re-evaluating as they mature.  But I would sleep better at night if I knew the Legacy themes won't be deleted :)

**Comment by jhpratt** [comment]
And just for the record — I sent an email to @equinusocio stating, among other things, that I would be willing to help maintain the theme.

**Comment by rianquinn** [comment]
WOW.... I opened my editor and it looked COMPLETELY different, and I did nothing to make it happen. I understand wanting to create a new theme, but why on earth did you do it under the existing extension.

There is no way this is more readable:
![image](https://user-images.githubusercontent.com/5932003/71854681-2743af00-309c-11ea-973a-0758f3c2c210.png)

Sorry... but I am going to have to switch to a different them. You could have at least warned us!!!

**Comment by maclockard** [comment]
@rianquinn Keep in mind the folks who maintain this do so for basically free out of the goodness of their hearts, no reason to be inflammatory. Myself and many others prefer the older theme, however, its important to remain constructive with how things can be improved and why.

**Comment by rianquinn** [comment]
I didn't say anything inflammatory... I simply expressed my frustration with having an existing extension take a 180 degree change out of what seems like nowhere. But I do agree... I could have provided some feedback on what I don't like... so here it is:
- The comments are no longer "in the background" as they are bright and blue. 
- The large amounts of white text against a dark background hurts my eyes
- The underlines suggest an error or misspelling and are really distracting. 
- In general, everything is either really dark, or really light, which again, just hurts to look at.

**Comment by sellsjonathan** [comment]
Hmm, honestly this was a little shocking to me.  I always loved how the, now, legacy themes separated elements of the syntax. It's gotten so much less specific. I hope legacy themes don't go away - or the new ones get expanded.

**Legacy Pale Night**
![image](https://user-images.githubusercontent.com/8787299/71857678-64b53600-30b6-11ea-96cd-cc76c2f35cf1.png)

**New Palenight**
![image](https://user-images.githubusercontent.com/8787299/71857704-7dbde700-30b6-11ea-8dd5-10dc1c1d4934.png)

**Comment by wSedlacek** [comment]
I was quite surprised to open my Editor and find the theme was quite different.
I agree with many of the others posing in here, the new theme is definitely less readable.

Let's take a look at some Angular TypeScript
Before:
![Screen Shot 2020-01-06 at 18 48 24](https://user-images.githubusercontent.com/8206108/71859670-5e728a00-30b5-11ea-80e5-c340f61e2026.png)

After:
![Screen Shot 2020-01-06 at 18 48 03](https://user-images.githubusercontent.com/8206108/71859675-65010180-30b5-11ea-86bb-47236fab96f8.png)

There are a few key points here that make this difficult for me to use
- The names of the variables are the same color as the commas, colons and brackets in the object.
- The red colors of the keywords `import` `export` and `from` are very deep red, and difficult for me to read. They also make me thing their is an error since I am typically scanning for red underlines
- The Decorator's `@` and the name of the decorator are very different colors breaking the connection between the two in my mind.

And now some HTML
Before:
![Screen Shot 2020-01-06 at 18 57 13](https://user-images.githubusercontent.com/8206108/71859958-74cd1580-30b6-11ea-8a31-ba57c733e0a2.png)

After:
![Screen Shot 2020-01-06 at 18 56 57](https://user-images.githubusercontent.com/8206108/71859950-6d0d7100-30b6-11ea-974e-b4453df4af35.png)

There again there are a few things that really bother me
- The bright blue for the tags while the `<` and `/>` are deep red really feels out of place to me. Especially when you contrast that against the dull purple.
- The syntax highlighting of `&amp;` is completely lost
- The syntax highlighting from `Angular HTML Syntax` seems to be entirely broken with the `{{ title }}`

I am not against changing the color scheme, however I think this may be a step back.
Thank you very much for all your hard work vsc-material-theme team! I really do appreciate your efforts. Thank you very much for reading my concerns as well.

**Comment by grig0ry** [comment]
Great update. I think it's time to find another theme.

**Comment by wSedlacek** [comment]
After Removing `Bracket Pair Colorizer` the highlighting in the TypeScript did become better, however I still believe there is still a lot of room for improvement.
![Screen Shot 2020-01-06 at 21 13 29](https://user-images.githubusercontent.com/8206108/71865798-9aafe580-30c9-11ea-80ca-50997f17c06a.png)

I have noticed a few additional things:
- The italics on the HTML attributes is now gone. I think I am gonna miss that even if I never noticed it before.
- Class Names have started to become underlined and popups are covered in underlines too. 
![Screen Shot 2020-01-06 at 21 17 28](https://user-images.githubusercontent.com/8206108/71865896-fbd7b900-30c9-11ea-8bfd-92432d6922b4.png)
- There are options to override the colors, however I fear with the rewrite much of the nuance may have been lost. I do hope some of these nuances return.

**Comment by vigzmv** [comment]
Thanks to all the contributors of material-theme, this has been my all-time fav.
But like the others here, I also prefer the old ones.
Is there any way I can switch back to the old ones on my vscode

**Comment by equinusocio** [comment]
Thank you for your feedbacks. We'll work to fix and improve the new schemes to improve readability in the next weeks.

**Comment by Omegastick** [comment]
The change for C++ is quite jarring IMO.

Old:
![old_scheme](https://user-images.githubusercontent.com/17452111/71874253-ccc25580-3164-11ea-96f9-89304d1d8712.png)
New:
![new_scheme](https://user-images.githubusercontent.com/17452111/71874257-d055dc80-3164-11ea-941e-ec0ae7d528d8.png)

As you can see, the syntax highlighting has gone pretty sharply downhill. Before, built-in types and keywords were distinct from user-defined names. Namespaces were also highlighted. Methods are more consistently highlighted now, though.

IMO, the old theme is beautifully understated, so I'll be sticking to the legacy themes for the time being anyway.

**Comment by yzhang-gh** [comment]
I would like to add one example about the comment color:

![image](https://user-images.githubusercontent.com/7588612/71875176-a994a780-315e-11ea-98a5-864a4f29aa40.png)

It would be better to not use such a bright color at the beginning of a comment (`%` here).

I guess this is an attempt to make comments more noticeable.
And that's why I am using double `%` for "important" comments (and also use another extension to make them bold).

Thanks for your great work.

**Comment by equinusocio** [comment]
@yzhang-gh the comment highlight was a bug and not intentional. We'll fix this.

**Comment by hymair** [comment]
I fell in love with Palenight (High Contrast) and the new look it completely different. And not in a good way, to be honest. I personally stick to a theme for a long time so I don't think it's a good idea to remove old ones completely. People get used to a theme and forget about it, and just focus on their work. Bringing this kind of breaking change isn't productive imo.

TL;DR; Yes, to new themes, but keep old ones

**Comment by Coding-Kiwi** [comment]
Not so much of a fan of the change either.. 
+1 for keeping Legacy in and I really appreaciate the hard work

Before:
![image](https://user-images.githubusercontent.com/5007641/71880004-a635f780-312f-11ea-9d9a-2a5c912fa6e1.png)

Now:
![image](https://user-images.githubusercontent.com/5007641/71880032-b77f0400-312f-11ea-9477-39259aa4cdfd.png)

Would it be possible to keep the Legacy themes instead of deleting them at some point in the future?

**Comment by GasimGasimzada** [comment]
I am going to throw in my example here as well:

![Screen Shot 2020-01-07 at 1 17 09 PM](https://user-images.githubusercontent.com/807231/71883517-076dc300-3150-11ea-983c-89c3771737c5.png)

I can barely distinguish between red and dark orange. Can we please have legacy themes stay intact?

**Comment by philippkuehn** [comment]
I also think that wasn't so good.

Vue components highlighting:

before:
![Bildschirmfoto 2020-01-07 um 10 39 55](https://user-images.githubusercontent.com/2500670/71885195-26f9f100-313a-11ea-9fa6-23979d83d7d1.png)

after:
![Bildschirmfoto 2020-01-07 um 10 39 11](https://user-images.githubusercontent.com/2500670/71885210-2cefd200-313a-11ea-83be-fc2362abb65a.png)

**Comment by Yago** [comment]
Sorry to add another bit of spices in this discussion, but even the legacy themes are “broken”.

Here is an example of (Legacy) **Ocean High Contrast** (which I love) before and after the last updates 😅 

<img width="1145" alt="Capture_d_écran_2020-01-07_10_42_19" src="https://user-images.githubusercontent.com/2025924/71885441-aab3dd80-313a-11ea-9926-d69467924dff.png">

**Comment by Jebble** [comment]
> The red/green pairing is something we can work on, but new themes are meant to be **new** and different from older schemes, they will never look the same but we can fix/improve color pairing with new schemes.

You should have created a new scheme altogether. Not change what we've been using for years. The update scared the shit out of me coming back from my holidays and I had no idea what was going on (no release notes shown by VS Code even..) 

If you're plannin gon deleting the legacy schemes please make sure there's an easy way for people to keep using those. I honestly can't imagine anyone liking this change, it's just unreadable.

**Comment by Jebble** [comment]
> Sorry to add another bit of spices in this discussion, but even the legacy themes are “broken”.
> 
> Here is an example of (Legacy) **Ocean High Contrast** (which I love) before and after the last updates 😅
> 
> <img alt="Capture_d_écran_2020-01-07_10_42_19" width="1145" src="https://user-images.githubusercontent.com/2025924/71885441-aab3dd80-313a-11ea-9926-d69467924dff.png">

Yours seem to be very usable. My new Ocean High Contrast looks like this
![image](https://user-images.githubusercontent.com/1397260/71888608-d043e580-3140-11ea-8bcb-c0cde2d3d211.png)

**Comment by pierreburton91** [comment]
As everyone out there, I almost fell off my chair today wondering "WTF happened?!".
Thanks for bringing back legacy themes under the Legacy flag. Just don't delete them.

**Don't fix it if it's not broken guys !**

**Comment by RichieAHB** [comment]
My colleagues at work this morning:

![tenor](https://user-images.githubusercontent.com/1652187/71891293-3844fc80-313e-11ea-865c-7447eaab7249.gif)

This feels like one of those things that, when you get used to _might_ be alright (although there are still a few issues to iron out, as mentioned above). That said, dropping it on people after the holidays, with this many changes, seems like a decision that comes from forgetting what a big change it is because you've been working on it / using it for ages.

FWIW it also must have updated at the same time as I installed another plugin and the change was so different I actually uninstalled the other plugin thinking it was that!

Either way, I'm sure there was no malice but I can empathise with those struggling their way into 2020 with this change! I'm gonna keep using it (in Legacy for now), which shows how much I love it and appreciate the work that's gone into it - so keep iterating, thanks, and happy new year!

**Comment by torbjornw** [comment]
![image](https://user-images.githubusercontent.com/5180706/71893512-3124eb00-314c-11ea-8818-195c966d2c6e.png)

This was a terrible change. I know that you are doing this for free and I dont have to use this extension. I was about to go back to the standard VScode-theme before I found the legacy-option still left. Then I had to google what the hell happened and I ended up here. 

From the screenshot above there are several issues:
1. Methodname and keyword return has the same color.
2. Parameters of method have the same color as named local variables 
3. Look at the "number|string", it looks like it is "number/string" because it is cursive. Extremely confusing...
4. Everything is just red and magenta. The red colors looks almost the same.

This is what I found in just that method (typescript).

I absolutely loved the old theme (Material Theme Ocean). Awesome work guys!
But these new themes should have been a new  extension or new themes?

I really hope that the legacy-themes are kept intact, otherwise I have to change color-theme...again...

Anyway thanks alot for the legacy-theme, I still love it!

**Comment by TheAifam5** [comment]
That is the "update"? I must say, the "new" color schemes are seriously ugly and even changing between new schemes, the syntax highlighting/colors looks in every case the same but the brightness changes.

Thats in TypeScript:
New
![image](https://user-images.githubusercontent.com/2192274/71896603-8238dd00-3154-11ea-8925-03ca33a6a9b5.png)

Old
![image](https://user-images.githubusercontent.com/2192274/71896625-967cda00-3154-11ea-8662-8384a5837332.png)


On C#:
![image](https://user-images.githubusercontent.com/2192274/71896721-cfb54a00-3154-11ea-8853-13d2d365b01f.png)

**Comment by equinusocio** [comment]
We got your feedbacks and we'll work on new schemes to improve readability, but since most of the things you've reported are based on personal taste (some of our users love new schemes someone don't) and since **we can't satisfy everyone's wishes**, we strongly suggest to:

- Keep using old schemes until they'll be deleted. 
- Check our readme to learn how to use VSCode and its API to [customise Material Theme](https://github.com/material-theme/vsc-material-theme#color-scheme-override) locally.
- Stop complaining and enjoy something that is brought to you **for free** thanks to who help us with donations.

**Comment by equinusocio** [comment]
Some fixes are already delivered. Thanks to all who **really** collaborated.

**Comment by jhpratt** [comment]
>     * Stop complaining and enjoy something that is brought to you **for free** thanks to who help us with donations.

Have you considered my offer of being a co-maintainer?

So far, all I've seen from you is numerous responses that are dismissive of all feedback, citing that it's freely available. While true, maintainers have a certain responsibility, which I presume you understand.

Keep in mind that you _are_ receiving funds to maintain this. If you're not willing to bring in another maintainer despite your complaints about only having two people, I (and others) won't hesitate to contact the companies and users sponsoring your work. They have a right to know how you're treating other users, and likely won't appreciate your breaking change and complete dismissal of respectful feedback.

**Comment by ACDR** [comment]
Seeing as @equinusocio seems set on the legacy themes being deleted at some point in the future, and that I'd rather not return to a completely different theme through no action of my own again any time soon, I've forked and published a new Legacy Material Theme as permitted by the Apache 2.0 license: https://marketplace.visualstudio.com/items?itemName=ACDR.vsc-legacy-material-theme

This can at least serve as a permanently accessible extension to maintain access to the legacy themes in the event that the legacy themes are deleted from this extension.

**Comment by equinusocio** [comment]
@ACDR You're toxic for the open source ecosystem. Instead of collaborating to the original repository, you made a new copy violating the license. Please close the repository or we'll start a DMCA.

      Redistribution. You may reproduce and distribute copies of the
      Work or Derivative Works thereof in any medium, with or without
      modifications, and in Source or Object form, provided that You
      meet the following conditions:

      (a) You must give any other recipients of the Work or
          Derivative Works a copy of this License; and

      (b) You must cause any modified files to carry prominent notices
          stating that You changed the files; and

      (c) You must retain, in the Source form of any Derivative Works
          that You distribute, all copyright, patent, trademark, and
          attribution notices from the Source form of the Work,
          excluding those notices that do not pertain to any part of
          the Derivative Works; and

**Comment by equinusocio** [comment]
@jhpratt Your email was not so "collaborative". First of all this thread is almost empty of feedback and full of complaints. A feedback is different, constructive and useful, saying "I don't like this" isn't a kind of feedbacks we want and we need (see other issues).

Then maintainers have the responsibility to make things working, not beautiful for everyone. Also, there are different kind of people in the OSS world, who love and trust open source, and who just spit shit on the work made by other.
